src/dependency-manager/src/manager.ts

```diff
@@ -580,5 +580,51 @@ export default abstract class DependencyManager {
     return res as ComponentConfig;
   }
 
-  abstract getArchitectContext(graph: DependencyGraph): Promise<any | undefined>;
+  protected generateEnvironmentIngresses(graph: DependencyGraph, components_map: Dictionary<ComponentConfig>): Dictionary<Dictionary<InterfaceSpec>> {
+    const all_external_interfaces: Dictionary<InterfaceSpec> = {};
+    for (const component_config of Object.values(components_map)) {
+      const component_interfaces = component_config.getInterfaces();
+      for (const [component_interface_name, interface_data] of Object.entries(component_interfaces)) {
+        if (!interface_data.host?.startsWith('${{')) {
+          all_external_interfaces[component_interface_name] = interface_data;
+        }
+      }
+    }
+
+    const ingresses: Dictionary<Dictionary<InterfaceSpec>> = {};
+    for (const [component_name, component_config] of Object.entries(components_map)) {
+      const component_interfaces = component_config.getInterfaces();
+      for (const component_interface_name of Object.keys(component_interfaces)) {
+        const tagless_component_name = component_config.getName();
+
+        let to_node = graph.getNodeByRef(`${component_name}-interfaces`);
+        let current_interface_name = component_interface_name;
+        let dependent_nodes = graph.getDependentNodes(to_node);
+        while (dependent_nodes.length) {
+          const dependent_node = dependent_nodes.pop();
+          const edge = graph.edges.find(edge => edge.from === dependent_node.ref && edge.to === to_node.ref);
+          if (edge instanceof IngressEdge && all_external_interfaces[current_interface_name]) {
+            if (!ingresses[tagless_component_name]) {
+              ingresses[tagless_component_name] = {};
+            }
+            ingresses[tagless_component_name][component_interface_name] = all_external_interfaces[current_interface_name];
+            break;
+          } else if (!dependent_nodes.length) {
+            dependent_nodes = graph.getDependentNodes(dependent_node);
+            to_node = dependent_node;
+            if (edge) {
+              const interface_mapping = Object.entries(edge.interfaces_map).find(([k, v]) => v === current_interface_name);
+              if (interface_mapping) {
+                current_interface_name = interface_mapping[0];
+              }
+            }
+          }
+        }
+      }
+    }
+
+    return ingresses;
+  }
+
+  abstract getArchitectContext(graph: DependencyGraph, components_map: Dictionary<ComponentConfig>): Promise<any | undefined>;
```

Comment:
You could probably make this not abstract and then you don't need to define in the local.